apis/hive/v1/clusterdeploymentcustomization_types.go

```diff
@@ -54,9 +57,50 @@ type PatchEntity struct {
 	// From is the json path to copy or move the value from
 	// +optional
 	From string `json:"from,omitempty"`
-	// Value is the value to be used in the operation
-	// +required
-	Value string `json:"value"`
+	// Value is the *string* value to be used in the operation. For more complex values, use
+	// ValueJSON.
+	// +optional
+	Value string `json:"value,omitempty"`
+	// ValueJSON is a string representing a JSON object to be used in the operation. As such,
+	// internal quotes must be escaped. If nonempty, Value is ignored.
+	// +optional
+	ValueJSON []byte `json:"valueJSON,omitempty"`
+}
+
+// Encode returns a string representation of the RFC6902 patching operation represented by the
+// PatchEntity.
+func (pe *PatchEntity) Encode() string {
```

Comment:
Does this function need the be exposed in our public API? It could just be another short standalone util function in util/yaml.go (`func encodePatchEntity(pe *PatchEntity) string`)

Reply:
It probably doesn't *need* to be exposed, no; but I did so because it's simple, doesn't pull in any extra deps, and could facilitate downstreams making use of PatchEntity for their own RFC6902ing.